### PR TITLE
feat: add iam role for services

### PR DESCRIPTION
**Pull request by jhyelton**
I've been testing this in [this](https://github.com/TRANSFR-Inc/vr-platform/pull/52/files) draft PR in vr-platform.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -31,4 +31,16 @@ variable "security_group_ids" {
   description = "Additional SG IDs to use for the service."
   type = list(string)
   default = []
+}
+
+variable "role_policy_arns" {
+  description = "List of ARNs of IAM policies to attach to IAM role"
+  default     = []
+  type        = list(string)
```
Comment (transfr-kennedy) [inline variables.tf:39]: If you change this from `list` to `set`, you can use this variable in the `for_each` meta-arg directly without converting

File: iam.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+resource "aws_iam_role_policy_attachment" "role_policy_arns" {
+  for_each   = toset(var.role_policy_arns)
+  role       = aws_iam_role.this.name
+  policy_arn = each.value
+}
+
+resource "aws_iam_policy" "this" {
+  for_each = var.role_policies
+  name        = "${var.app_name}-${var.env_name}-${each.key}"
+  path        = "/app/"
```
Comment (transfr-kennedy) [inline iam.tf:27]: Why is this path different than that of the role above?

File: iam.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  policy      = each.value
+}
+ 
+resource "aws_iam_role_policy_attachment" "this" {
```
Comment (transfr-kennedy) [inline iam.tf:31]: This _could_ be combined with the other above if you wanted. I am for separating the two though if it provides value.

**Review by transfr-kennedy — APPROVED** [review]
Qs and optional style suggestions, but LGTM overall! :shipit:

File: iam.tf (hunk 2 above)
Comment (jhyelton) [reply iam.tf:27]: I copied these paths/code [from the implementation in PlatformOrchestration](https://github.com/TRANSFR-Inc/PlatformOrchestration/blob/develop/.tx/tf/main.tf), but I'll make them consistent.

**Pull request merged**